WordPress/Sniffs/WP/DeprecatedParameterValuesSniff.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPress\Sniffs\WP;
+
+use WordPress\AbstractFunctionParameterSniff;
+
+/**
+ * Check for usage of deprecated parameter values in WP functions and provide alternative based on the parameter passed.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   1.0.0
+ *
+ * @uses    \WordPress\Sniff::$minimum_supported_version
+ */
+class DeprecatedParameterValuesSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * The group name for this group of functions.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var string
+	 */
+	protected $group_name = 'wp_deprecated_parameter_values';
+
+	/**
+	 * Array of function, argument, and replacement function for deprecated argument.
+	 *
+	 * The list of deprecated parameter values can be found by
+	 * looking for `_deprecated_argument()`, `apply_filters_deprecated()`
+	 * and `do_action_deprecated()`.
+	 * The list is sorted first by the WordPress version when deprecated
+	 * and then alphabetically.
+	 * Last updated for WordPress 4.9.6.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *    $target_functions = array(
+	 *        (string) Function name. => array(
+	 *            (int) Target parameter position, 1-based. => array(
+	 *                (string) Parameter value. => array(
```

Comment:
This sub-array setup will only work if the expected type for the value passed is a text string token or an integer.

How is this sniff going to handle deprecated parameter values of another type ? Like booleans, floats etc.

N.B.: the format of the complete array is "free" with the only requirement being the function names being the top-level keys. You don't have to follow the examples set by other sniffs for the array format if it's not the best format for this sniff.

Reply:
> How is this sniff going to handle deprecated parameter values of another type ? Like booleans, floats etc.

I am trying to think of code example where this would be needed. I came up with the following test but I can't think of a situation where this would be needed.

```php
function bar_foo( $value ) {
	if ( false === $value ) {
		// Deprecated. Use String instead.
	}
}
```